### PR TITLE
[gazebo_ros_control] Indigo compatibility

### PR DESCRIPTION
**Pull request by adolfo-rt**
These changesets require ros-controls/ros_control#156.

### CODE REVIEW

**Comment by davetcoleman** [comment]
+1 reviewed.

**Comment by adolfo-rt** [comment]
Please don't merge this PR yet.

@davetcoleman You might be interested in testing these changes once ros-controls/ros_control#156 is merged. I have only checked that they build so far. I might have time to do a bit more next week. I'm not a user of the default plugin, and there's no test suite to validate against, so actual user validation would be valuable.

I tried to adapt the `DefaultRobotHWSim` implementation to support transmissions with multiple joints and multiple hardware interfaces, but realized that the class needs an important refactoring and didn't continue. I opened #186 to keep track of this issue, which is important in its own right.

File: gazebo_ros_control/package.xml [hunk 1]
```diff
@@ -27,7 +26,6 @@
   <build_depend>urdf</build_depend>
   
   <run_depend>roscpp</run_depend> 
-  <run_depend>gazebo_ros</run_depend>
```
Comment (scpeters) [inline gazebo_ros_control/package.xml:30]: I think it still needs the `run_depend` on gazebo_ros (see [gazebo_ros_control_plugin.cpp:74-80](https://github.com/pal-robotics/gazebo_ros_pkgs/blob/gazebo-ros-control-indigo/gazebo_ros_control/src/gazebo_ros_control_plugin.cpp#L74-L80)).

Comment (davetcoleman) [inline gazebo_ros_control/package.xml:30]: Agreed.

**Comment by adolfo-rt** [comment]
PR modified to preserve the runtime dependency on `gazebo_ros`.

**Comment by adolfo-rt** [comment]
The pending ros_control for Indigo PR has been merged. Yet ros_control and friends have not yet been released into Indigo, only `control_msgs` is. Let's get that done and then merge this one.

**Comment by davetcoleman** [comment]
I created the branches for it yesterday, but someone else needs to test it on an indigo machine.

**Comment by scpeters** [comment]
It builds for me on saucy/indigo, but I haven't run any code. Any recommendations for testing?

**Comment by adolfo-rt** [comment]
I would say that this tutorial is the best there is now, would you agree @davetcoleman?

http://gazebosim.org/wiki/Tutorials/1.9/ROS_Control_with_Gazebo

**Comment by davetcoleman** [comment]
Agreed, not the best documentation on non-simulation

sent from my phone

dave :: 251.463.2345

**Comment by scpeters** [comment]
Sorry I haven't managed to do any run-time testing. Since it builds for me, I would propose merging this so that we can make releases and then resolve any issues as they are reported.

+1

**Comment by adolfo-rt** [comment]
OK, I'm merging. Building from source should work, but take into account that the ros-controls repos have not yet been released into the buildfarm for indigo. It's on my todo list. As of today, all critical PRs on the ros-controls organization for indigo are either merged or submitted.

**Comment by scpeters** [comment]
Thanks @adolfo-rt !

**Pull request merged**